libs/community/langchain_community/chat_models/anyscale.py

```diff
@@ -102,22 +101,17 @@ def get_available_models(
 
         return {model["id"] for model in models_response.json()["data"]}
 
-    @root_validator(pre=True)
-    def validate_environment_override(cls, values: dict) -> dict:
+    @root_validator()
+    def validate_environment(cls, values: dict) -> dict:
```

Comment:
why was this necessary?

Reply:
(the change in the validator type)

Reply:
I recalled that we needed this change, otherwise it would complain no `openai_api_token` environment set

Reply:
Just tested and confirmed. with pre=True, you will get following validation error

 "Did not find openai_api_key, please add an environment variable `OPENAI_API_KEY` which contains it, or pass `openai_api_key` as a named parameter. (type=value_error)"

Reply:
My intuition is that this will still fail if you don't have `OPENAI_API_KEY` in your environment then. I think this is the reason this implementation used to store the api key in `openai_api_key` instead of `anyscale_api_key`. It may be worth it to leave that as-is.

Reply:
User should NOT use OPENAI_API_KEY environment var when using Anyscale, but only need to supply ANYSCALE_API_KEY. This change is for this purpose. Otherwise code would fail even if you have ANYSCALE_API_KEY but missing OPENAI_API_KEY